docs/raw/push-notification-server.md

```diff
@@ -0,0 +1,672 @@
+be sent with `error` set to `VERSION_MISMATCH`.
+
+If any other error occurs the `error` should be set to `INTERNAL_ERROR`.
+
+If the request is successful, an access token MUST be generated and sent back to the client
+with `success` set to `true`.
+
+It is RECOMMENDED to use a randomly generated string of at least 36 characters.
+
+Otherwise a response MUST be sent with `success` set to `false`.
+
+`request_id` should be set to the `SHA3` of the signature sent by the client.
+
+The request MUST be sent on the [partitioned topic][./2-account.md] of the sender.
+
+The payload of the response is:
+
+```protobuf
+message PushNotificationPreferencesAcknowledgement {
+  boolean success = 1;
+  ErrorType error = 2;
+  bytes request_id = 3;
+  PushNotificationPreferences preferences = 4;
+  string access_token = 5;
+
+  enum ErrorType {
+    UNKNOWN_ERROR_TYPE = 0;
+    MALFORMED_MESSAGE = 1;
+    VERSION_MISMATCH = 2;
+    UNSUPPORTED_TOKEN_TYPE = 3;
+    INTERNAL_ERROR = 4;
+  }
+}
+```
+
+[//]: (We can ratched and use a partitioned topic here)
+
+A client SHOULD listen for an acknowledgement sent on their [partitioned topic][./2-account.md].
+
+If `success` is `true` the client has registered successfully.
+
+If `success` is `false`:
+
+- If `MALFORMED_MESSAGE` is returned, the request SHOULD not be retried without ensuring 
+  that is correctly formed.
+- If `INTERNAL_ERROR` is returned, the request MAY be retried, but the client MUST be 
+  backoff exponentially
+- If `VERSION_MISMATCH` is returned, the client SHOULD ensure that the remote version returned in `preferences` is integrated, and the version should be incremented and the request MAY be sent again.
+
+A client MAY register with multiple Push Notification Servers in order to increase availability.
+
+A client SHOULD make sure that all the notification services they registered with have the same information about their devices and tokens.
+
+If no response is returned the request SHOULD be considered failed and MAY be retried with the same server or a different one, but it MUST exponentially backoff after each trial.
+
+If the request is successful a token is returned. This SHOULD be 
+[advertised](#advertising-a-push-notification-server) as described below if no public key filtering is necessary.
+
+## Re-registering with the push notification server
+
+A client SHOULD re-register with the node if the APS or FIREBASE token changes.
+
+When re-registering a client SHOULD ensure that it has the most up-to-date
+`PushNotificationPreferences`, update the part relative to their `installation_id`
+if necessary, increment `version` and send a `PushNotificationRegister` as described above.
+
+Once re-registered, a client SHOULD advertise the changes.
+
+
+## Changing options
+
+This is handled in exactly the same way as re-registering above.
+
+## Unregistering from push notifications
+
+To unregister a client MUST send a `PushNotificationRegister` request as described
```

Comment:
Should this be explicit in terms of 'when a user logs out' the client should deregister?

I suppose that is part of a the main client spec, not here

Reply:
Though is not on logout (otherwise you will stop receiving notifications), you will want to unregister if you don't want to receive notifications anymore, or don't want to use this server anymore.

Reply:
Hm, it's not clear to me what the correct design is here. If you logout as one user and login as a new user, you could argue that the first user shouldn't receive notifications anymore. To me that might be more intuitive, than getting PNs for some user user.

Use case: family sharing a phone, or some kind of kiosk mode where you shouldn't receive PNs for other suers.

Reply:
I double checked with UX and indeed is how they think it should be behave, until explicit logout you should receive notifications.
It is a bit of a UX concern, so not sure if it's needed to be specified here.
For now I will add it as a MAY and the implementation in status-react will reflect this behavior.

Reply:
Fair, agre re deferring it.

@hesterbruikman IMO this UX decision should be revisited. It goes against expectations and also explicitly against previous PN designs I have made, and just sounds off to me. If you are logged off you should NOT receive PNs IMO, that's what that action does. I doubt it is how other IMs work either tbh.